### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by PhilipVinc** [netket/netket#559]
I would like to keep here a list of issues and enhancements to consider for the future

New Features:
 - Add dynamics
 - Add Quantum State Reconstruction
 - Support gradient of non-hermitian operators (the code is already implemented, but there might be bugs and it's disabled)
 - Implement gradient of expectation values of operators on mixed states

Models:
 - Add Symmetric Dense layer to have RBMSymm #555
 - Add Autoregressive FFNN

Hilbert:
 - Find a better way to work with `DoubledHilbert` and to construct it. 
 - Allow a mechanism to override the `random_state` and `flip_site` for certain doubled hilbert, such as homogeneous ones. They don't need to go through the `jax.lax.switch` all the time

Samplers:
 - Autodetect CPUs/GPUs and decide to which sampler to dispatch
 - Use Numba in samplers #554 
 
Tests:
 - Add benchmarks to avoid performance regressions #68 
 - Eventually consider benchmarking also jut-compile time?
 - Properly test Liouvillian gradient (right now it's tested against the legacy code) 

Misc
- Possibly improve jit time of `nk.sampler.sample` by refactoring the code and where we jit
- Keep an eye out for serialising jax jitted functions: [Jax#476](https://github.com/google/jax/issues/476)
- Upstream what we need for complex number support to Flax and eventually remove `nk.nn`
- Some utils to deserialise our `.log` files and reconvert to our history objects and make plotting easier might be nice. 

Issues:
 - `MCState.samples` before and after serialization/deserialization don't match. Right now, when serialising a variational state, we store the parameters and the `sampler_state`. However, if we have already sampled the state, the current `sampler_state` will generate the next batch of samples. So we should keep around the last `sampler_state` and serialise that one, so that when calling `.samples` after deserialisation we get the same samples.

Docs:
 - Docs have plenty of build warnings. Those should be addressed
 - Add Developer documentation
 - Maybe move the core of the docs (not the website) to ReadTheDocs for automatic versioning?

**Comment by gcarleo** [comment]
Yes, I would add 

- Remove Legacy
